src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.MultiSegment.Tests.cs

```diff
@@ -0,0 +1,283 @@
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleEmptySingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleEmptySingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_MultipleEmptySingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_multipleEmptySingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleShortSingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleShortSingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_MultipleShortSingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_multipleShortSingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleLongSingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleLongSingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        private static void MultiSegment_GenericReadTest(ReadOnlySequence<byte> data, JsonCommentHandling commentHandling)
+        {
+            for (int i = 0; i < Iterations; i++)
+            {
+                var state = new JsonReaderState(options: new JsonReaderOptions { CommentHandling = commentHandling });
+                var json = new Utf8JsonReader(data, isFinalBlock: true, state);
+
+                while (json.Read())
+                {
+                }
+            }
+        }
+
+        private static ReadOnlySequence<byte> GetSequence(byte[] dataUtf8, int segmentSize)
+        {
+            int numberOfSegments = dataUtf8.Length / segmentSize + 1;
+            byte[][] buffers = new byte[numberOfSegments][];
+
+            for (int j = 0; j < numberOfSegments - 1; j++)
+            {
+                buffers[j] = new byte[segmentSize];
+                Array.Copy(dataUtf8, j * segmentSize, buffers[j], 0, segmentSize);
+            }
+
+            int remaining = dataUtf8.Length % segmentSize;
+            buffers[numberOfSegments - 1] = new byte[remaining];
+            Array.Copy(dataUtf8, dataUtf8.Length - remaining, buffers[numberOfSegments - 1], 0, remaining);
+
+            return CreateReadOnlySequence(buffers);
+        }
+
+        private static ReadOnlySequence<byte> CreateReadOnlySequence(params byte[][] buffers)
+        {
+            if (buffers.Length == 1)
+                return new ReadOnlySequence<byte>(buffers[0]);
+            var list = new List<Memory<byte>>();
+            foreach (var buffer in buffers)
```

Comment:
nit: don't use var here

Reply:
this is copy paste of corefx code

Reply:
Yep, I know. It was an oversight there too and we should fix it (also was in test code). It was likely missed in review.